### PR TITLE
Add Makefile for Lambdas

### PR DESCRIPTION
**Pull request by kenoir**
💻  🌭

### CODE REVIEW

File: lambdas/Makefile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+ROOT = $(shell git rev-parse --show-toplevel)
+
+ifneq ($(ROOT), $(shell pwd))
+	include $(ROOT)/shared.Makefile
+endif
+
+
+.docker/_lambda_deps: .docker/python3.6_ci
```
Comment (alexwlchan) [inline lambdas/Makefile]: I think you want this to be `$(ROOT)/.docker/_lambda_deps`, so it always goes to `.docker` in the root of the repo, regardless of whether you run `make` from root or from within this directory.

Also, `.docker/terraform_ci` and `.docker/python3.6_ci` and maybe `uptodate-git` need to move into shared.Makefile.

File: lambdas/Makefile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	docker run -v $$(pwd)/lambdas:/data -e OP=install-deps python3.6_ci:latest
+
+## Run tests for our Lambda code
+lambdas-test: .docker/python3.6_ci
+	./scripts/run_docker_with_aws_credentials.sh -v $$(pwd)/lambdas:/data -e OP=test -e FIND_MATCH_PATHS='./*/target common/tests' python3.6_ci:latest
```
Comment (alexwlchan) [inline lambdas/Makefile]: Consider breaking onto multiple lines?

File: lambdas/Makefile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+## Run a plan on lambda stack
+terraform-lambda-plan: uptodate-git .docker/terraform_ci .docker/_lambda_deps
+	docker run -v $$(pwd)/terraform:/terraform -v $$(pwd)/lambdas:/data -v $$HOME/.aws:/root/.aws -v $$HOME/.ssh:/root/.ssh -e OP=plan terraform_ci:latest
+
+## Run an apply on lambda stack
+terraform-lambda-apply: .docker/terraform_ci
+	docker run -v $$(pwd)/terraform:/terraform -v $$(pwd)/lambdas:/data -v $$HOME/.aws:/root/.aws -v $$HOME/.ssh:/root/.ssh -e OP=apply terraform_ci:latest
+
+
+.PHONY: terraform-lambda-plan terraform-lambda-apply loris-publish
```
Comment (alexwlchan) [inline lambdas/Makefile]: loris-publish? 🤔

File: lambdas/README.md [hunk 4]
```diff
@@ -1,15 +1,13 @@
...
-.. code-block:: console
-
-   $ cd /path/to/platform-infra
-   $ cd lambdas/name-of-lambda
-   $ ln -s ../common/sns_utils.py .
+```
+$ cd /path/to/platform-infra
+$ cd lambdas/name-of-lambda
+$ ln -s ../common/sns_utils.py .
+```
```
Comment (alexwlchan) [inline lambdas/README.md]: I don’t think this is accurate any more.

File: lambdas/Makefile (hunk 1 above)
Comment (kenoir) [reply lambdas/Makefile]: 👍  but boing to leave `uptodate-git` as it doesn't quite feel like ti fits the pattern of waht we're putting in the `shared.Makefile` at the moment.

File: lambdas/Makefile [hunk 5]
```diff
@@ -0,0 +1,36 @@
+ROOT = $(shell git rev-parse --show-toplevel)
+
+ifneq ($(ROOT), $(shell pwd))
+include $(ROOT)/shared.Makefile
+endif
+
+
+.docker/_lambda_deps: $(ROOT)/.docker/python3.6_ci
+	docker run -v $$(pwd)/lambdas:/data -e OP=install-deps python3.6_ci:latest
```
Comment (alexwlchan) [inline lambdas/Makefile]: This should be `$(ROOT)`

File: lambdas/README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# lambdas
+
+This directory contains the  AWS Lambdas pacakged up as a single "service".
+
+## Overview
+
+The lambdas in this directory comprise supporting functions for a number of the other services in this project. Ultimately they should be moved into the service they support rathr than being represented here.
```
Comment (alexwlchan) [inline lambdas/README.md]: tyop: /s/rathr/rather

File: shared.Makefile [hunk 7]
```diff
@@ -1,5 +1,11 @@
 # This Makefile contains tasks and underlying pieces that are shared
 # across our other Makefiles.
+#
+# Build the Docker images used for CI tasks.
+#
+# The script sticks a record that the image has been built in .docker, so the
+# image isn't rebuilt unless you run 'make clean' first.  This makes CI tasks
+# slightly less chatty when run locally.
```
Comment (alexwlchan) [inline shared.Makefile]: 👍 

useful comment

I can imagine pushing these some of them down into their individual projects as appropriate, but this is fine for now.

File: shared.Makefile [hunk 8]
```diff
@@ -16,3 +22,26 @@ $(ROOT)/.docker/publish_service_to_aws:
...
+	mkdir -p .docker && touch .docker/_build_deps
+
+$(ROOT)/.docker/miro_adapter_tests:
+	./scripts/build_ci_docker_image.py \
+		--project=miro_adapter_tests \
+		--dir=miro_adapter \
+		--file=miro_adapter/miro_adapter_tests.Dockerfile
+
+clean:
+	rm -rf .docker
```
Comment (alexwlchan) [inline shared.Makefile]: should be `$(ROOT)/.docker`

File: lambdas/Makefile [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+## Run tests for our Lambda code
+lambdas-test: $(ROOT)/.docker/python3.6_ci
+	./scripts/run_docker_with_aws_credentials.sh \
```
Comment (kenoir) [inline lambdas/Makefile]: Must reference $(ROOT)

**Pull request merged**